sentry-android-core/src/main/java/io/sentry/android/core/ANRWatchDog.java

```diff
@@ -73,16 +83,39 @@ public void run() {
       }
 
       // If the main thread has not handled ticker, it is blocked. ANR.
-      if (tick.get() != 0 && !reported) {
-        //noinspection ConstantConditions
+      if (tick.get() != 0 && !reported.get()) {
         if (!reportInDebug && (Debug.isDebuggerConnected() || Debug.waitingForDebugger())) {
           logger.log(
               SentryLevel.DEBUG,
               "An ANR was detected but ignored because the debugger is connected.");
-          reported = true;
+          reported.set(true);
           continue;
         }
 
+        // we only raise an ANR event if the process is in ANR state.
+        // if ActivityManager is not available, we'll still be able to send ANRs
+        final ActivityManager am =
```

Comment:
we could add a flag here to keep the old behavior, so we don't check its process, not sure if it's worth though.
this flag would live in `SentryAndroidOptions` and enabled by default.

Reply:
Agree not sure it's worth it. Let's take this approach as the new default and see how it goes first